00_download_data/nbconverted/0_download_data.py

```diff
@@ -72,14 +73,39 @@
 
 md5_sum = get_ipython().getoutput('md5sum $me_filepath')
 print(md5_sum[0])
-assert md5_sum[0].split(' ')[0] == manifest_df.loc['methylation'].md5
+assert md5_sum[0].split(' ')[0] == manifest_df.loc['methylation_27k'].md5
 
 
-# ### Download RPPA data
+# ### Download 450k probe DNA methylation data (warning: large file, ~40GB)
+# 
+# This took me overnight (~12 hours) to download, although it could be faster on some connections...
 
 # In[7]:
 
 
+me_id, me_filename = manifest_df.loc['methylation_450k'].id, manifest_df.loc['methylation_450k'].filename
+url = 'http://api.gdc.cancer.gov/data/{}'.format(me_id)
+me_filepath = os.path.join(cfg.raw_data_dir, me_filename)
+
+if not os.path.exists(me_filepath):
+    urlretrieve(url, me_filepath)
+else:
+    print('Downloaded data file already exists, skipping download')
+
+
+# In[8]:
+
+
+md5_sum = get_ipython().getoutput('md5sum $me_filepath')
```

Comment:
Ah, interesting. Python has the `hashlib` package that you might already know, but using `md5sum` seems more direct and easy (although it might not work on Windows or macOS; I think in macOS the command is slightly different).

Reply:
Yeah, I remember looking into this before - I think using `hashlib` to get file hashes is a bit more complicated and not super efficient for large files (see, e.g. [this post](https://stackoverflow.com/a/16876405)), so I decided that just piping the file straight to `md5sum` would be easier.

Good to know that it might not work on other platforms, though. I'll make a note in the script.